### ISSUE TITLE
eCommerce Onboarding: No new WooCommerce Products are created

### ISSUE THREAD

**Issue by sixhours** [Automattic/wp-calypso#71158]
1. Go to `/setup/ecommerce`
2. Run through the flow, choosing the Thriving Artist theme.
3. Note there are Products listed in the Thriving Artist Headstart annotation: `/premium/thriving-artist/inc/headstart/en.json`
4. Note there are WooCommerce Products shown on the Thriving Artist demo site: https://thrivingartistdemo.wpcomstaging.com
5. Create your new site.
6. Visit the Products tab in your Dashboard; you won't see any Products listed.

@mreishus noted that the same flow with the same theme from `/setup` works as expected, so it's likely something is messed up with Headstart in the eCommerce flow specifically.

**Comment by mreishus** [comment]
When I make a new site using the ecommerce flow these steps happen in order:
* (A) It creates the site
* (B) It headstarts the site to `pub/lettre` (presumably because I have not yet paid for an ecommerce plan and therefore do not have access to premium themes)
* (C) I checkout and buy ecommerce
* (D) It transfers the site to atomic

We need to add a step between (C) and (D) where it headstarts the site to `premium/theme-name-here` before starting the atomic transfer.

Example workflow of a bundled theme that works (not using the ecomm flow): D94413-code#1962621 ( D94413-code ).

**Comment by mreishus** [comment]
The flow that we've tested the most is [stepper default flow -> choose tsubaki or thriving artist -> stepper plugin bundle flow]. In these cases, the sites end up with the products on them, and everything is working well and looking nice.

I've tried to make the ecommerce flow emulate these steps in D95250-code , but something is still "off" currently. For example, the site's "Shop" page sends me to "Comissions".
